File: WealthWise.AI/UserProfile.py
from pydantic import BaseModel, Field, ConfigDict
from typing import List, Optional
import uuid
from RiskToleranceLevel import RiskToleranceLevel as RiskTolerance
from Investment import Investment
from FinancialGoal import FinancialGoal

class UserProfile(BaseModel):
    model_config = ConfigDict(
        populate_by_name=True,
        use_enum_values=True,
        arbitrary_types_allowed=True
    )
    
    Id: Optional[str] = Field(default_factory=lambda: str(uuid.uuid4()), alias="Id")
    Name: str = Field(..., alias="Name")
    Email: str = Field(..., alias="Email")
    Age: Optional[int] = Field(0,ge=18, le=100, alias="Age")
    MonthlyIncome: Optional[float] = Field(0.0, alias="MonthlyIncome")
    MonthlySavings: Optional[float] = Field(0.0, alias="MonthlySavings")
    RiskTolerance: Optional[str] = Field(RiskTolerance.Conservative, alias="RiskTolerance")
    CurrentInvestments: Optional[List[Investment]] = Field(default_factory=list, alias="CurrentInvestments")
    FinancialGoals: Optional[List[FinancialGoal]] = Field(default_factory=list, alias="FinancialGoals")

File: WealthWise.Shared/Models/UserProfile.cs
using System.ComponentModel.DataAnnotations;

namespace WealthWise.Shared.Models;

public class UserProfile
{
    public UserProfile(string name, string email, int age, decimal monthlyIncome, decimal monthlySavings, RiskToleranceLevel riskToleranceLevel, List<Investment> currentInvestments, List<FinancialGoal> financialGoals)
    {
        Name = name;
        Email = email;
        Age = age;
        MonthlyIncome = monthlyIncome;
        MonthlySavings = monthlySavings;
        RiskTolerance = riskToleranceLevel.ToString();
        CurrentInvestments = currentInvestments ?? [];
        FinancialGoals = financialGoals ?? [];
    }

    public Guid Id { get; set; }
    [Required]
    public string Name { get; set; } = string.Empty;
    [Required]
    [EmailAddress]
    public string Email { get; set; } = string.Empty;

    [Required]
    [Range(18, 100)]
    public int Age { get; set; }
    public decimal MonthlyIncome { get; set; }
    public decimal MonthlySavings { get; set; }
    public RiskToleranceLevel RiskToleranceLevel { get; set; }
    public string RiskTolerance { get; set; } = string.Empty;
    public List<Investment> CurrentInvestments { get; set; } = new();
    public List<FinancialGoal> FinancialGoals { get; set; } = new();

    public UserProfile() { }
}

public enum RiskToleranceLevel
{
    Conservative,
    Moderate,
    Aggressive
}